### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ahmedkareem999**
Please describe the purpose of this pull request below.
The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)

Copy/paste from CHANGELOG.md is good

### Added 

- name in FundRanking.sol

- a function that converts from string to bytes32 in shares.sol

- fundRanking.js test file

- PicopsComplaince.sol

- picopsComplaince.js test file

### Changed

- function name in FundRanking.sol

- Changed Subscription to Investment.

- Changed contract to interface in ComplianceInterface.sol

### Fixed 

- Lint and Test

### CODE REVIEW

File: src/compliance/ComplianceInterface.sol [hunk 1]
```diff
@@ -3,7 +3,7 @@ pragma solidity ^0.4.19;
 /// @title Compliance Interface Contract
 /// @author Melonport AG <team@melonport.com>
 /// @notice This is to be considered as an interface on how to access the underlying Compliance Contract
-interface ComplianceInterface {
```
Comment (travs) [inline src/compliance/ComplianceInterface.sol]: Just wondering if we can change back `contract` to `interface`, or if there is a reason why this change was made.
We are trying to use `interface` where possible now.

File: tests/integration/fundRanking.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+let fundRanking;
+let deployed;
+let version;
+let manager;
+let deployer;
+
+const fundNames = ["Fund Name 1", "Fund Name 2"];
+
+// To convert bytes to string
+function binToString(array){
```
Comment (travs) [inline tests/integration/fundRanking.js]: Nice!

File: tests/integration/fundRanking.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    const convertedChar = (String.fromCharCode(bytes[i]));
+    if (convertedChar === "\u0000") {
+      break;
+    }
+		result+= convertedChar;
+	}
+	return result;
+}
+
+// To return values from the array
```
Comment (travs) [inline tests/integration/fundRanking.js]: Not quite sure what this function is doing.
Since it is only called once, maybe we can get rid of it and just destructure the result of `getFundDetails` inside the unit test itself.
What do you think?

File: tests/integration/fundRanking.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+       [deployed.SimpleAdapter.address],
+       v,
+       r,
+       s,
+     ],
+   );
+ });
+
+ // test to check getFundDetails() method
+ test('get address, shareprice, time and name of all funds in a version', async (t) => {
```
Comment (travs) [inline tests/integration/fundRanking.js]: Good :+1:

File: tests/unit/compliance/picopsCompliance.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    "compliance/PicopsCompliance",
+    {from: deployer},
+    [simpleCertifier.address]
+  );
+});
+
+test('Checks if investment is permitted', async (t) => {
+  accounts = await api.eth.accounts();
+  const beforeInvestmentPermitted = await picopsCompliance.instance
+    .isInvestmentPermitted.call({}, [accounts[1], 1000000000000000000, 1000000000000000000]);
```
Comment (travs) [inline tests/unit/compliance/picopsCompliance.js]: using `accounts[1]` was a good change here :+1:

File: src/assets/Shares.sol [hunk 6]
```diff
@@ -35,6 +35,14 @@ contract Shares is Asset, SharesInterface {
...
 
     function getName() view returns (string) { return name; }
+
+    // Function to convert from string to bytes32
+    function getNameinBytes32() view returns (bytes32 result) {
+        bytes memory localName = bytes(name);
+        assembly {
+          result := mload(add(localName, 32))
+        }
+    }
```
Comment (retotrinkler) [inline src/assets/Shares.sol]: is this needed?

Comment (ahmedkareem999) [reply src/assets/Shares.sol]: Yes, I used that to convert from string to bytes32 so that I can get the name in bytes32. In funds.sol the name is returned in the form of bytes32. So i had to change that in shares.sol.

Comment (retotrinkler) [reply src/assets/Shares.sol]: why did u need the name in bytes32?

Comment (ahmedkareem999) [reply src/assets/Shares.sol]: Because the string can't be used in funds.sol, I had to use bytes32.

Comment (retotrinkler) [reply src/assets/Shares.sol]: Think u meant FundRanking.sol

Comment (ahmedkareem999) [reply src/assets/Shares.sol]: Sorry, yes!

**Comment by GauthamN** [comment]
Merged manually in 22fa4dbfededd6ec3494f277552be8098751d96b
